Stop emitting conditions for Swift 5.5-era features into textual interfaces

DougGregor opened the pull request:
The "#if compiler(>=5.3) && $AsyncAwait" checks were necessary for staging in concurrency in Swift 5.5. At this point, it's safe to assume that any compiler that tries to read a generated Swift interface file will support concurrency, so we can stop emitting these guards.

DougGregor committed:
Stop emitting conditions for Swift 5.5-era features into textual interfaces

The "#if compiler(>=5.3) && $AsyncAwait" checks were necessary for
staging in concurrency in Swift 5.5. At this point, it's safe to assume
that any compiler that tries to read a generated Swift interface file will
support concurrency, so we can stop emitting these guards.

DougGregor commented:
@swift-ci please smoke test

tshortli commented on lib/AST/ASTPrinter.cpp:
```diff
@@ -2963,34 +2963,10 @@ static bool usesFeatureStaticAssert(Decl *decl) {
 }
 
 static bool usesFeatureAsyncAwait(Decl *decl) {
-  if (auto func = dyn_cast<AbstractFunctionDecl>(decl)) {
```
It could be nice to have a brief comment in these `usesFeature` functions that explains that we're returning `false` because the features are old.

DougGregor replied:
I was thinking of adding a macro locally that documented this better, e.g.,

```swift
ASSUME_FEATURE_IN_ALL_SWIFT_COMPILERS(AsyncAwait)
```

tshortli replied:
Yeah that seems like good option too.

DougGregor replied:
I went with the `BASELINE_LANGUAGE_FEATURE` approach in `Features.def`

tshortli commented on test/ModuleInterface/actor_availability.swift:
```diff
@@ -40,36 +38,34 @@ public actor UnavailableActor {
 extension Enum {
-  // CHECK: #if compiler(>=5.3) && $Actors
-  // CHECK-NEXT: @_hasMissingDesignatedInitializers public actor ExtensionNestedActor {
+  // CHECK-NOT: #if compiler(>=5.3) && $Actors
```
Nit: In all of these tests there was some effort made to align what the `CHECK` lines were matching for visual clarity. It would be nice if that were preserved.

DougGregor replied:
Alright, I've cleaned these up a bit

DougGregor committed:
Macro-metaprogram "baseline" features that all Swift compilers handle

This is better than a pile of usesXXXFeature() functions that return
`false` with no explanation.

DougGregor committed:
Clean up formatting in test cases

DougGregor commented:
@swift-ci please smoke test
